src/predis/mock.clj

```diff
@@ -287,6 +290,19 @@
       (when (<= idx last-idx)
         (get vs idx'))))
 
+  (core/linsert [this k pos pivot v]
+    (assert (or (= pos "before") (= pos "after")) err-syntax)
+    (let [vs (vec (core/get this k))]
+      (if (not (seq vs))
+        0
+        (if (contains? (set vs) pivot)
+          (let [[before after] (split-at ((if (= pos "before") util/id inc)
+                                            (.indexOf vs pivot)) vs)
+                vs' (vec (flatten (concat before [(str v)] after)))]
```

Comment:
`concat` should be returning a flat seq here, the `flatten` step should be unnecessary and I'd bet the `vec` step is unecessary too